src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -403,31 +403,79 @@ private static string GetParameterText(string parameterName)
 
                 CommandProcessorBase commandProcessor = null;
                 CommandRedirection[] commandRedirection = null;
-                for (int i = 0; i < pipeElements.Length; i++)
-                {
-                    commandRedirection = commandRedirections != null ? commandRedirections[i] : null;
-                    commandProcessor = AddCommand(pipelineProcessor, pipeElements[i], pipeElementAsts[i],
-                                                  commandRedirection, context);
-                }
 
-                var cmdletInfo = commandProcessor?.CommandInfo as CmdletInfo;
-                if (cmdletInfo?.ImplementingType == typeof(OutNullCommand))
-                {
-                    var commandsCount = pipelineProcessor.Commands.Count;
-                    if (commandsCount == 1)
+                var pipelineAst = (PipelineAst)pipeElementAsts[0].Parent;
+                if (pipelineAst.BackgroundProcess)
+                {
+                    // For background jobs rewrite the pipeline as a Start-Job command
+                    var scriptblockBodyString = ((PipelineAst)pipeElementAsts[0].Parent).Extent.Text;
+                    Token[] tokens;
+                    ParseError[] errors;
+                    var pipelineBodyAst = Parser.ParseInput(scriptblockBodyString, out tokens, out errors);
+                    // Prefix variables in the scriptblock with $using: 
+                    var variables = pipelineBodyAst.FindAll(x => x is VariableExpressionAst, true);
+                    System.Text.StringBuilder updatedScriptblock = new System.Text.StringBuilder();
```

Comment:
Yes, but `StringBuilder` starts with 16 characters and adds chunks by the larger of how much you are appending and how big the buffer is already.

With many small appends, you'll probably end up with multiple chunks, whereas starting with the minimum needed will result in probably 1 extra allocation, or you could just double the minimum needed if it's "small" (say less than 40kb or so) to avoid a single allocation going into the LOH.